plugins/scaffolder/src/plugin.ts

```diff
@@ -47,6 +51,31 @@ export const scaffolderPlugin = createPlugin({
   },
 });
 
+export const RepoUrlPicker = scaffolderPlugin.provide(
+  createScaffolderFieldExtension({
+    // TODO: work out how to type this component part so we can enforce FieldComponent from RJSF
+    component: RepoUrlPickerComponent,
+    name: 'RepoUrlPicker',
+    validation: (value: JsonValue, validation) => {
```

Comment:
👌 

Guess we should we also try to tie together the `onChange` type with the validation `value` too? Making `createScaffolderFieldExtension` generic.